iceoryx_hoofs/test/moduletests/test_cxx_spinator.cpp

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2022 by Apex.AI Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "test.hpp"
+using namespace ::testing;
+
+#include "iceoryx_hoofs/internal/cxx/spinator.hpp"
+
+using namespace iox::cxx::internal;
+
+namespace
+{
+TEST(SpinatorTest, repeatCallingYieldTheUntilItSaturatesTakesAtLeast10ms)
+{
+    class SpinatorSut : public spinator
+    {
+      public:
+        using spinator::INITIAL_REPETITIONS;
+    };
+
+    constexpr uint64_t REPETITIONS = SpinatorSut::INITIAL_REPETITIONS;
+    spinator sut;
+
+    auto start = std::chrono::steady_clock::now();
+    for (uint64_t i = 0U; i < REPETITIONS; ++i)
+    {
+        sut.yield();
+    }
+    auto end = std::chrono::steady_clock::now();
+
+    constexpr iox::units::Duration MIN_RUNTIME = iox::units::Duration::fromMilliseconds(10);
+
+    EXPECT_THAT(std::chrono::nanoseconds(end - start).count(), Ge(MIN_RUNTIME.toNanoseconds()));
```

Comment:
Do we want a test that verifies that we sleep `FINAL_WAIT_TIME` after `INITIAL_REPETITIONS`? This would be some simple test that shows that the sleep time increases.